### ISSUE TITLE
WiFi Documentation

### ISSUE THREAD

**Issue by stevemac00** [nodemcu/nodemcu-firmware#2456]
Make sure you read and understand http://nodemcu.readthedocs.io/en/dev/en/support/.
Use one of the two templates below and **DELETE THE REST**.

8<------------------------ BUG REPORT -----------------------------------------
### Expected behavior
This is more of a documentation request but the wifi developers might also be interested.

**Background:** I upgraded 8 modules from various manufacturers to 2.2 and noticed that my modules would disconnect MQTT causing a LWT broadcast then reconnect again. This would happen several times each hour. Everything still worked as it should but I wanted to dig in to learn why it was happening with 2.2 and not the 1.X version.

The MQTT Client went offline because the wifi did a STA LEAVE on my AP. The AP logs were full of LEAVE/JOIN/DHCP requests.

Most of my modules have very low RSSI (-70 to -85) although my noise floor runs about -96 and I'm in a remote detached home with low interference.

Upon learning the WiFi disconnects I relocated one of the previously failed modules close enough to the AP to get a -45 RSSI and it never disconnected over several days. I surmised this was due to some change to the RF environment but it turns out to be more complicated.

After looking through the documentation there were some new wifi calls not available in 1.5. When I looked at wifi.getcountry() it was set for CN (China). Also, the wifi.sta.sleeptype() was set to wifi.MODEM_SLEEP. I updated these to US and wifi.NONE_SLEEP and the disconnects disappeared.

I don't know why the one module started working flawlessly (no LEAVE/JOIN) when the RSSI improved and to the naive it doesn't seem related.

### Test code
```Lua
function heartbeat()
  local wifiConnected=wifi.sta and wifi.sta.status()==wifi.STA_GOTIP
  if mwifiConnected~=wifiConnected then
    print('wifi:'..tostring(wifiConnected)..' reconnects='..tostring(wifiConnects))
    mwifiConnected=wifiConnected
    if wifiConnected then wifiConnects=wifiConnects+1 end
  elseif not wifiConnected and tmr.time()%3==0 then
    print(uart.write(0,'.'))
  end
 end
wifiConnects=0
heartbeatTmr=tmr:create()
heartbeatTmr:register(500,tmr.ALARM_AUTO,heartbeat)
heartbeatTmr:start()
```
### NodeMCU version
2.2 master

### Hardware
Various

8<------------------------ END BUG REPORT -------------------------------------

**Comment by marcelstoer** [comment]
@stevemac00 that is a lot of interesting information, thanks for sharing. We made `getcountry`/`setcountry` available with 2.2. and mentioned them in the release notes at https://github.com/nodemcu/nodemcu-firmware/releases/tag/2.2.0-master_20180402.

> This is more of a documentation request

What exactly do you expect us to document? AFAICS everything you described is accurately documented in the [WiFi module API documentation](https://nodemcu.readthedocs.io/en/latest/en/modules/wifi/), isn't it? What (extra) documentation would have your life easier?

@dnc40085 (WiFi module maintainer) and @devsaurus (worked on `setcountry`) what's your take on this?

**Comment by stevemac00** [comment]
Let me preface by saying you, Terry and the others have done a magnificent job and 2.2 is solid and a massive improvement.

Perhaps a warning to set sleeptype() and setcountry() because the defaults could change application behavior.

My internet search found no similar problems with low RSSI causing STA:LEAVE/JOIN so maybe I have some unique combinations. (My AP's are all Ubiquiti 2.4GHz with legacy B turned off and 20MHz g/n channels.)

Perhaps this thread is sufficient documentation in case anyone comes across this in a search.
